test/integration_test/storkctl_schedulepolicy_test.go

```diff
@@ -0,0 +1,196 @@
+	t.Run("createDefaultWeeklySchedulePolicyTest", createDefaultWeeklySchedulePolicyTest)
+	t.Run("createDefaultMonthlySchedulePolicyTest", createDefaultMonthlySchedulePolicyTest)
+	t.Run("createCustomIntervalSchedulePolicyTest", createCustomIntervalSchedulePolicyTest)
+	t.Run("createCustomDailySchedulePolicyTest", createCustomDailySchedulePolicyTest)
+	t.Run("createCustomWeeklySchedulePolicyTest", createCustomWeeklySchedulePolicyTest)
+	t.Run("createCustomMonthlySchedulePolicyTest", createCustomMonthlySchedulePolicyTest)
+	t.Run("deleteSchedulePolicyTest", deleteSchedulePolicyTest)
+	err = setRemoteConfig("")
+	require.NoError(t, err, "setting kubeconfig to default failed")
+}
+
+func createSchedulePolicyTest(t *testing.T, policyType string, args map[string]string, specFileName string) {
+	factory := storkctl.NewFactory()
+	schedulePolicyName := "automation-test-policy"
+	var outputBuffer bytes.Buffer
+	defer SchedulePolicyCleanup(t, schedulePolicyName)
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	cmdArgs := []string{"create", "schedulepolicy", schedulePolicyName, "-t", policyType}
+	//add the custom args to the command if any
+	for key, value := range args {
+		cmdArgs = append(cmdArgs, "--"+key)
+		cmdArgs = append(cmdArgs, value)
+	}
+	cmd.SetArgs(cmdArgs)
+	//execute the command
+	if err := cmd.Execute(); err != nil {
+		logrus.Infof("Execute storkctl failed: %v", err)
+		return
+	}
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	fmt.Printf("Actual output is: %s\n", actualOutput)
+	expectedOutput := fmt.Sprintf("Schedule policy %v created successfully\n", schedulePolicyName)
+	require.Equal(t, expectedOutput, actualOutput)
+
+	//Validate the created resource
+	specFile := "specs/storkctl-specs/schedulepolicy/" + specFileName
+	err := ValidateSchedulePolicyFromFile(t, specFile, schedulePolicyName)
+	require.NoError(t, err, "Error trying to validate the created resource")
+
+	// If we are here then the test has passed
+	testResult := testResultPass
+	logrus.Infof("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+func deleteSchedulePolicyTest(t *testing.T) {
+	factory := storkctl.NewFactory()
+	schedulePolicyName := "delete-test-policy"
+	schedulePolicy := &storkv1.SchedulePolicy{
+		ObjectMeta: meta.ObjectMeta{
+			Name: schedulePolicyName,
+		},
+		Policy: storkv1.SchedulePolicyItem{
+			Monthly: &storkv1.MonthlyPolicy{
+				Date: 30,
+				Time: "12:15PM",
+			},
+		},
+	}
+	_, err := storkops.Instance().CreateSchedulePolicy(schedulePolicy)
+	if err != nil {
+		logrus.Errorf("unable to create schedule policy %v", schedulePolicyName)
+	}
+
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	cmdArgs := []string{"delete", "schedulepolicy", schedulePolicyName}
+	cmd.SetArgs(cmdArgs)
+	if err := cmd.Execute(); err != nil {
+		logrus.Infof("Execute storkctl failed: %v", err)
```

Comment:
Lets print the whole command in case of failure.

Reply:
Addressed in the revision